server/src/lib.rs

```diff
@@ -330,38 +360,37 @@ async fn mirror<'reader>(
         .map(|x| x.to_str().unwrap().contains("gzip"))
         .unwrap_or(false);
     let request: Result<serde_json::Value, _> = serde_json::from_slice(&body);
-    let (geth_response_parts, geth_response_bytes, parsed_geth_response) =
-        match proxy(path, query, method, headers.clone(), body, port).await {
-            Ok(response) => {
-                let (parts, body) = response.into_parts();
-                let raw_bytes = hyper::body::to_bytes(body)
-                    .await
-                    .expect("Failed to get geth response");
-                let bytes = if is_zipped {
-                    match try_decompress(&raw_bytes) {
-                        Ok(x) => x,
-                        Err(e) => {
-                            println!("WARN: gz decompression failed: {e:?}");
-                            let body = hyper::Body::from(raw_bytes);
-                            return Ok(Response::from_parts(parts, body));
-                        }
-                    }
-                } else {
-                    raw_bytes.to_vec()
-                };
-                match serde_json::from_slice::<serde_json::Value>(&bytes) {
-                    Ok(parsed_response) => (parts, raw_bytes, parsed_response),
-                    Err(_) => {
-                        println!("Request: {:?}", &request);
-                        println!("headers: {headers:?}");
-                        println!("WARN: op-geth non-json response: {:?}", bytes);
-                        let body = hyper::Body::from(bytes);
+    let parsed_geth_response = match proxy(path, query, method, headers.clone(), body, port).await {
```

Comment:
No need to `proxy` any more, or at least we can call it something else. Originally, right at the beginning, all the calls were proxied to `op-geth`. That's where this function name comes from.